### PR TITLE
Unit tests for some of the SecureChannel module functions

### PR DESCRIPTION
**Pull request by Infinity95**
This PR is aimed at creating unit tests for the SecureChannel module that specifically test just this module.
For that purpose i created a testing SecurityPolicy which can be found in the testing-plugins folder.
The testing policy also contains assertions, that generally check if the functions are called correctly.
Please leave comments with suggestions if you have any.

See also: #1296

### CODE REVIEW

**Comment by Pro** [comment]
@Infinity95 you can now rebase on master, the other PR is merged

**Comment by Infinity95** [comment]
Is it possible to disable the errors that currently occur in travis? I find them to be really annoying since i don't want to write an extra assert makro (or even cast every single pointer that is asserted) just so the error disappears. The nullpointer dereference analysis is also a false positive in my opinion, since the pointer that is dereferenced is checked beforehand to not be null.

**Comment by jpfr** [comment]
You can just write
```c
ck_assert(securityPolicy != NULL);
```

Or, if you absolutely have to prevent a warning, add `-Wnoincompatible-pointer-types-discards-qualifiers` to the compiler settings in `/tests/CMakeLists.txt`. The settings here normally do not leak into the parent directory.

In general, we should use conersative settings everywhere. Otherwise we get lazy and the code quality decreases.

**Comment by Infinity95** [comment]
Is there a way to avoid the analyzer warnings without having to add an if(xy != NULL) check to everything? I mean the dereference can never happen, since the assertion kills the program before that, or am i overlooking something?

**Comment by Infinity95** [comment]
Shouldn't we exclude the tests and testing plugins from the coverage (if that's possible)?
They make the coverage seem better than it is i think.

**Comment by jpfr** [comment]
> Shouldn't we exclude the tests and testing plugins from the coverage (if that's possible)?

👍 For more honesty in our numbers.

**Comment by Infinity95** [comment]
How do i exclude them from coverage then? I saw that the examples were excluded, but where is it done?

**Comment by Infinity95** [comment]
See #1375

**Comment by Infinity95** [comment]
@Pro can you help me out with this? It seems something is going wrong with the amalgamation in travis but i can't figure out why or what is causing it. Locally on my machine everything is fine (tried multiple build configurations).

**Comment by Infinity95** [comment]
Ok seems i fixed it. My ide added an include with <> brackets which caused the amalgamation to fail on travis and appveyor, but not locally. I don't know why lol

**Comment by Infinity95** [comment]
@jpfr Can you merge this batch of unit tests after everything is green? I'd like to do these tests in smaller batches, so that we get some measures against regression without having to wait until every single test is done.

**Comment by jpfr** [comment]
Hey @Infinity95 
Can you rebase on master? We can then merge.

**Comment by jpfr** [comment]
Hey @Infinity95 ,

I rebase the branch on master. Can you give me push access to this branch in your repo?

**Comment by Infinity95** [comment]
Done

**Comment by jpfr** [comment]
Thx. Rebased.

I had to disable one unit test.
Some .hex files are missing.
E.g. ${CMAKE_CURRENT_SOURCE_DIR}/dumps/client_HELOPN.hex

**Comment by jpfr** [comment]
@Pro Did you remove ${CMAKE_CURRENT_SOURCE_DIR}/dumps/client_HELOPN.hex?
That is, don't we need that anymore since we autogenerate the corpus?

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -154,7 +160,56 @@ endif()
 # Readspeed server
 add_executable(check_server_readspeed server/check_server_readspeed.c $<TARGET_OBJECTS:open62541-object> $<TARGET_OBJECTS:open62541-testplugins>)
 target_link_libraries(check_server_readspeed ${LIBS})
-add_test_valgrind(server_readspeed ${TESTS_BINARY_DIR}/check_server_readspeed)
+add_test_valgrind(check_server_readspeed ${TESTS_BINARY_DIR}/check_server_readspeed)
+
+# Test server with network dumps from files
```
Comment (Pro) [inline tests/CMakeLists.txt]: Remove these lines, up to line 212

**Pull request merged**